src/client/autosuggest.ts

```diff
@@ -0,0 +1,209 @@
+  protected readonly url = '/autosuggest';
+  protected readonly method = 'get';
+
+  public static init(
+    apiKey?: string,
+    config?: ApiClientConfiguration,
+    transport?: Transport
+  ): AutosuggestClient {
+    return new AutosuggestClient(apiKey, config, transport);
+  }
+
+  protected query(options: AutosuggestOptions) {
+    this.lastReqOpts = options;
+    return {
+      ...this.autosuggestOptionsToQuery(options),
+      input: options.input,
+    };
+  }
+
+  protected async validate(options: AutosuggestOptions) {
+    const textOptions = [AutosuggestInputType.Text];
+    const speechOptions = [
+      AutosuggestInputType.GenericVoice,
+      AutosuggestInputType.NMDP_ASR,
+      AutosuggestInputType.VoconHybrid,
+    ];
+    let valid = !!options;
+    let message: string | undefined = undefined;
+
+    if (!valid) {
+      message = 'You must provide at least options.input';
+      return { valid, message };
+    }
+    if (options.input.length < 1) {
+      valid = false;
+      message = 'You must specify an input value';
+    }
+    if (options.clipToCountry?.filter(country => country.length > 2)?.length) {
+      valid = false;
+      message =
+        'Invalid clip to country. All values must be an ISO 3166-1 alpha-2 country code';
+    }
+    if (
+      options.clipToBoundingBox &&
+      (options.clipToBoundingBox.southwest.lat >
+        options.clipToBoundingBox.northeast.lat ||
+        options.clipToBoundingBox.southwest.lng >
+          options.clipToBoundingBox.northeast.lng)
+    ) {
+      valid = false;
+      message =
+        'Southwest lat must be less than or equal to northeast lat and southwest lng must be less than or equal to northeast lng';
+    }
+    if (options.clipToPolygon) {
+      if (
+        !Array.isArray(options.clipToPolygon) ||
+        options.clipToPolygon.length < 4 ||
+        options.clipToPolygon.length > 25
+      ) {
+        valid = false;
+        message =
+          'Invalid clip to polygon value. Array must contain at least 4 coordinates and no more than 25';
+      }
+      const lastIndex = options.clipToPolygon.length - 1;
+      if (
+        options.clipToPolygon[0].lat !== options.clipToPolygon[lastIndex].lat ||
+        options.clipToPolygon[0].lng !== options.clipToPolygon[lastIndex].lng
+      ) {
+        valid = false;
+        message =
+          'Invalid clip to polygon value. The polygon bounds must be closed.';
+      }
+    }
+    if (options.inputType) {
+      if (![...textOptions, ...speechOptions].includes(options.inputType)) {
+        valid = false;
+        message =
+          'Invalid input type provided. Must provide a valid input type.';
+      }
+      if (
+        options.language === undefined &&
+        speechOptions.includes(options.inputType)
+      ) {
+        valid = false;
+        message = 'You must provide language when using a speech input type';
+      }
+    }
+    if (options.language && options.language.length > 2) {
```

Comment:
What if the language length is less than 2? Probably just easier to check that it _is_ 2.